### PR TITLE
WIP: Wit sample tests, using DomainMap to change image tag

### PR DESCRIPTION
**Pull request by hzhao-github**
oracle.kubernetes.operator.ItImageTool testCustomImageUsed SUCCESS in Three Jenkins jobs:

http://wls-jenkins.us.oracle.com/job/weblogic-kubernetes-operator-javatest/3920/consoleFull

http://wls-jenkins.us.oracle.com/job/weblogic-kubernetes-operator-javatest/3926/consoleFull

http://wls-jenkins.us.oracle.com/job/weblogic-kubernetes-operator-javatest/3927/consoleFull

### CODE REVIEW

File: integration-tests/src/test/java/oracle/kubernetes/operator/ItImageTool.java [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+  @BeforeAll
+  public static void staticPrepare() throws Exception {
+    if (FULLTEST) {
+      testClassName = new Object() {
+      }.getClass().getEnclosingClass().getSimpleName();
+
+      // Determine image name and version to be used
+      // load app props defined
+      LoggerHelper.getLocal().log(Level.INFO, "Loading props from: " + APP_PROPS_FILE);
+      Properties appProps = TestUtils.loadProps(APP_PROPS_FILE);
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/kubernetes/operator/ItImageTool.java]: You may want to take a look at BaseTest.initialize() method where we load the props file and have getter methods for the properties. Loading props file here is redundant.

File: integration-tests/src/test/java/oracle/kubernetes/operator/ItImageTool.java [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+   * to create WebLogic domain in Operator env.
+   * There are two ways to use the WLS Docker image created by imagetool
+   *  1. export IMAGE_NAME_WEBLOGIC = "name of your wls Docker image"
+   *     export IMAGE_TAG_WEBLOGIC = "version of the image"
+   *  2. use the values of weblogicImageName and weblogicImageTag in OperatorWIT.properties
+   *
+   * @throws Exception exception
+   */
+  @Test
+  public void testCustomImageUsed() throws Exception {
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/kubernetes/operator/ItImageTool.java]: Modify test to delete the domain and operator if the test completed successfully.

File: integration-tests/src/test/java/oracle/kubernetes/operator/utils/Domain.java [hunk 3]
```diff
@@ -860,7 +860,12 @@ public void createDomainOnExistingDirectory() throws Exception {
               + "\"";
       LoggerHelper.getLocal().log(Level.INFO,
           "making sure the domain directory exists by running " + cmd);
-      ExecResult result = TestUtils.exec(cmd);
+      // Looking for servers directory as sometimes krun.sh exits with non-zero
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/kubernetes/operator/utils/Domain.java]: these changes are already in develop branch, not sure why they are appearing in your PR. You may want to sync to latest develop branch and resolve conflicts.

File: integration-tests/src/test/java/oracle/kubernetes/operator/ItImageTool.java (hunk 1 above)
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/kubernetes/operator/ItImageTool.java]: Yea. I noticed that. However, I need props from this file before BaseTest.initialize()  gets called, plus I don't want to hard-code anything. That's why this call here

File: integration-tests/src/test/java/oracle/kubernetes/operator/utils/Domain.java (hunk 3 above)
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/kubernetes/operator/utils/Domain.java]: Just synced with latest develop branch

File: integration-tests/src/test/java/oracle/kubernetes/operator/ItImageTool.java (hunk 2 above)
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/kubernetes/operator/ItImageTool.java]: fixed

**Comment by hzhao-github** [comment]
The test passed in Jenkins jobs using latest changes:

http://wls-jenkins.us.oracle.com/job/weblogic-kubernetes-operator-javatest/3946/console
http://wls-jenkins.us.oracle.com/job/weblogic-kubernetes-operator-javatest/3947/console
http://wls-jenkins.us.oracle.com/job/weblogic-kubernetes-operator-javatest/3949/console
http://wls-jenkins.us.oracle.com/job/weblogic-kubernetes-operator-javatest/3950/console

**Comment by vanajamukkara** [comment]
@hzhao-github Can you update with external Jenkins results. You may need to sync to develop branch before running on external Jenkins as Ryan merged some changes on Friday.

**Comment by hzhao-github** [comment]
sure. I will

**Comment by hzhao-github** [comment]
it's synced up with latest develop b ranch:

bash-4.2$ git checkout develop
Branch develop set up to track remote branch develop from origin.
Switched to a new branch 'develop'
bash-4.2$ git pull
Already up-to-date.

**Comment by hzhao-github** [comment]
After synced up with latest develop branch, the test passed in internal Jenkins

http://wls-jenkins.us.oracle.com/job/weblogic-kubernetes-operator-javatest/3969/console

**Comment by vanajamukkara** [comment]
just fyi, waiting for external jenkins clean run ..please don't merge @rjeberhard

**Comment by hzhao-github** [comment]
@vanajamukkara No worries. I will not ask for merge until I get a clean run that includes this test on External Jenkins

**Comment by hzhao-github** [comment]
two more tests passed on Internal Jenkins

using 12.2.1.3.0 installer:

http://wls-jenkins.us.oracle.com/job/weblogic-kubernetes-operator-javatest/4000/consoleFull

using 12.2.1.4.0

http://wls-jenkins.us.oracle.com/job/weblogic-kubernetes-operator-javatest/4001/consoleFull

**Comment by bsmoyers** [comment]
Can one of the admins verify this patch?

**Comment by hzhao-github** [comment]
It passed on External Jenkins:

http://build.weblogick8s.org:8080/job/weblogic-kubernetes-operator-quicktest/948/consoleFull
http://build.weblogick8s.org:8080/job/weblogic-kubernetes-operator-quicktest/949/consoleFull

And on Internal Jenkins

http://wls-jenkins.us.oracle.com/job/weblogic-kubernetes-operator-javatest/4111/consoleFull
http://wls-jenkins.us.oracle.com/job/weblogic-kubernetes-operator-javatest/4118/console

**Comment by marinakog** [comment]
242 files changed, are you sure you need to merge all changes including Ryan's ?

**Comment by hzhao-github** [comment]
Thanks Marina to pointing out that. I didn't notice it :( I guess I need to start a new PR with latest develop branch

**Pull request closed without merging**